cub/test/catch2_test_warp_scan.cu

```diff
@@ -286,14 +302,18 @@ c2h::host_vector<T> compute_host_reference(
   {
     for (int w = 0; w < num_warps; ++w)
     {
-      T* output = result.data() + w * logical_warp_threads;
-      T current = initial_value;
-      for (int i = 0; i < logical_warp_threads; i++)
+      T* output     = result.data() + w * logical_warp_threads;
+      T accumulator = output[0];
+      T current     = static_cast<T>(scan_op(initial_value, output[0]));
+      output[0]     = current;
+      for (int i = 1; i < logical_warp_threads; i++)
       {
-        current   = static_cast<T>(scan_op(current, output[i]));
-        output[i] = current;
+        T tmp       = output[i];
+        current     = static_cast<T>(scan_op(current, tmp));
+        accumulator = static_cast<T>(scan_op(accumulator, output[i]));
+        output[i]   = current;
```

Comment:
Suggestion: for readability I would either write one of the following version. Otherwise, the code suggests like you would want to save the value of `output[i]` before you overwrite it **and** use the old value again later. But no writes happen to `output[i]` while its value is needed for reading.
```suggestion
        current     = static_cast<T>(scan_op(current, output[i]));
        accumulator = static_cast<T>(scan_op(accumulator, output[i]));
        output[i]   = current;
```
or
```suggestion
        T tmp       = output[i];
        current     = static_cast<T>(scan_op(current, tmp));
        accumulator = static_cast<T>(scan_op(accumulator, tmp));
        output[i]   = current;
```
or
```suggestion
        T& tmp      = output[i];
        current     = static_cast<T>(scan_op(current, tmp));
        accumulator = static_cast<T>(scan_op(accumulator, tmp));
        tmp         = current;
```

Reply:
makes absolute sense. thanks